packages/connect-wallet-modal/src/connectButtons/ConnectInjected.tsx

```diff
@@ -0,0 +1,61 @@
+import { FC, useCallback } from 'react';
+import { useConnect, useDisconnect } from 'wagmi';
+import { ConnectButton } from '../components';
+import { capitalize } from '../helpers';
+import { ConnectInjectedProps } from './types';
+
+export const ConnectInjected: FC<ConnectInjectedProps> = (
+  props: ConnectInjectedProps,
+) => {
+  const {
+    onConnect,
+    onBeforeConnect,
+    setRequirements,
+    metrics,
+    walletId,
+    walletName,
+    icon,
+    ...rest
+  } = props;
+
+  const walletIdCapitalized = capitalize(walletId);
+  const metricsOnConnect =
+    metrics?.events?.connect?.handlers[`onConnect${walletIdCapitalized}`];
+  const metricsOnClick =
+    metrics?.events?.click?.handlers[`onClick${walletIdCapitalized}`];
+
+  const { connectAsync, connectors } = useConnect({
+    onSuccess() {
+      onConnect?.();
+      metricsOnConnect?.();
```

Comment:
It's probably not the part of the PR, but it's always bugged me that we are bringing lido-specific metrics to open source repo 🤔  Feels like this kind of callbacks should be just handled by onConnect

Reply:
@kozlovzxc-at-lido 
The problem is that for now, `onConnect` is a common callback for all the buttons: the same `onConnect` will fire for different buttons, because of legacy. `metrics` was a workaround for this, it is not lido specific, it is just an empty callback. But of course we should make better callbacks.